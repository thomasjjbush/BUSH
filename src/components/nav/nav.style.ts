import styled from 'styled-components';
import { StyledProps } from '../../types';

export const StyledNav = styled.nav<StyledProps>`
    background-color: ${(props): string => props.theme.colors.highlight};
    box-shadow: 0px 3px 5px 0px rgba(50, 50, 50, 0.25);
    display: flex;
    justify-content: space-between;
    height: 60px;
    padding: 10px;
    z-index: 1;

    button {
        margin: 0;

        @media only screen and (max-width: ${(props): number => props.theme.breakpoints.mobile}px) {
            display: none;
        }
    }
`;

export const StyledActions = styled.div`
    align-items: center;
    display: flex;

    a {
        margin-right: 10px;

        i {
            color: ${props => props.theme.colors.text};
        }
    }
`;
