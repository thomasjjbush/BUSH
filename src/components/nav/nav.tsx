import React, { FunctionComponent, ReactElement, useCallback } from 'react';
import { useTheme } from 'styled-components';
import { Link } from 'react-router-dom';
import { Button, Icon, Image } from '../';
import { StyledNav, StyledActions } from './nav.style';
import { Buttons, Icons, Theme, NavProps as Props } from './../../types';

import darkModeCV from './../../cv/thomas-bush-cv-dark-mode.pdf';
import lightModeCV from './../../cv/thomas-bush-cv-light-mode.pdf';

const social = [
    {
        icon: Icons.LINKEDIN,
        testID: 'linkedin-icon',
        title: 'LinkedIn',
        url: 'https://www.linkedin.com/in/thomas-bush-555604161/',
    },
    { icon: Icons.GITHUB, testID: 'github-icon', title: 'Github', url: 'https://github.com/thomasjjbush' },
    { icon: Icons.WHATSAPP, testID: 'whatsapp-icon', title: 'WhatsApp', url: 'tel:+447501274429' },
];

export const Nav: FunctionComponent<Props> = ({ darkMode }: Props): ReactElement => {
    const { images } = useTheme() as Theme;
    const onClick = useCallback(() => window.open(darkMode ? darkModeCV : lightModeCV, '_blank'), [darkMode]);

    return (
        <StyledNav>
            <Link to="/">
                <Image alt="BUSH" src={images.logo.pink} size={200} height={40} />
            </Link>
            <StyledActions>
                {social.map(({ icon, testID, title, url }, i) => (
                    <a
                        data-test-id={testID}
                        key={i}
                        href={url}
                        rel="noreferrer"
                        title={title}
                        {...(!url.startsWith('tel:') && { target: '_blank' })}
                    >
                        <Icon icon={icon} size={30} />
                    </a>
                ))}
                <Button design={Buttons.PRIMARY} onClick={onClick}>
                    Download CV
                </Button>
            </StyledActions>
        </StyledNav>
    );
};

Nav.displayName = 'Nav';
