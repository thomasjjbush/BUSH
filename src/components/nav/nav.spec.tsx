import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './nav';

describe('Nav', () => {
    it('should match snapshot', () => {
        expect(shallow(<Nav darkMode={true} />)).toMatchSnapshot();
    });
});
