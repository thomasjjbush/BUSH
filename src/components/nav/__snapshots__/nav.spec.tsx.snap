// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<styled.nav
  styles="
    background-color: theme.colors.highlight;
    box-shadow: 0px 3px 5px 0px rgba(50, 50, 50, 0.25);
    display: flex;
    height: 60px;
    justify-content: space-between;
    padding: 10px;
    z-index: 1;

    button {
        margin: 0;

        @media only screen and (max-width: theme.breakpoints.mobilepx) {
            display: none;
        }
    }
"
>
  <Link
    to="/"
  >
    <Image
      alt="BUSH"
      height={40}
      size={200}
      src="theme.images.logo.pink"
    />
  </Link>
  <styled.div
    styles="
    align-items: center;
    display: flex;

    a {
        margin-right: 10px;

        i {
            color: theme.colors.text;
        }
    }
"
  >
    <a
      data-test-id="linkedin-icon"
      href="https://www.linkedin.com/in/thomas-bush-555604161/"
      key="0"
      rel="noreferrer"
      target="_blank"
      title="LinkedIn"
    >
      <Icon
        icon="eac9"
        size={30}
      />
    </a>
    <a
      data-test-id="github-icon"
      href="https://github.com/thomasjjbush"
      key="1"
      rel="noreferrer"
      target="_blank"
      title="Github"
    >
      <Icon
        icon="eab0"
        size={30}
      />
    </a>
    <a
      data-test-id="whatsapp-icon"
      href="tel:+447501274429"
      key="2"
      rel="noreferrer"
      title="WhatsApp"
    >
      <Icon
        icon="ea93"
        size={30}
      />
    </a>
    <Button
      design={0}
      onClick={[Function]}
    >
      Download CV
    </Button>
  </styled.div>
</styled.nav>
`;
